### PR TITLE
Tool: add a tool to configure sink

### PR DESCRIPTION
**Pull request by GwendalRaoul**
This tool is available with wm-node-conf
It can locally configure a node through DBUS (with a sink service attached)
but can also be used to list sinks config

### CODE REVIEW

**Comment by PFigs** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- python_transport/wirepas_gateway/configure_node.py  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/wirepas/gateway/pullRequest?prid=5235269&bid=17161461)

**Comment by PFigs** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- python_transport/wirepas_gateway/configure_node.py  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/wirepas/gateway/pullRequest?prid=5235269&bid=17161461)

**Comment by PFigs** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- python_transport/wirepas_gateway/configure_node.py  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/wirepas/gateway/pullRequest?prid=5235269&bid=17161461)

File: python_transport/wirepas_gateway/configure_node.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+    Helper to manipulate and convert node role based on a Base role
+    and a list of flags
+
+    """
+
+    class BaseRole(Enum):
+        SINK = 0x1
+        ROUTEUR = 0x2
+        NON_ROUTEUR = 0x3
```
Comment (vgeneves) [inline python_transport/wirepas_gateway/configure_node.py]: Not sure it is usefull but do you want to add advertiser role ?

File: python_transport/wirepas_gateway/configure_node.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+#
+import binascii
+import argparse
+
+from enum import Enum
+
+from wirepas_gateway.dbus.dbus_client import BusClient
+
+
+class NodeRole:
```
Comment (vgeneves) [inline python_transport/wirepas_gateway/configure_node.py]: What happens if you set an invalid combination (like Sink autorole) does this fail at some point with a clear error message ?

File: python_transport/wirepas_gateway/configure_node.py (hunk 1 above)
Comment (GwendalRaoul) [reply python_transport/wirepas_gateway/configure_node.py]: Yes:

```
$ wm-node-conf set -s sink0 -r "sink autorole"
Cannot set 129 for param node_role on sink sink0: GDBus.Error:com.wirepas.sink.config.error: [WPC_set_role]: C Mesh Lib ret = 4
Configuration done with result = GatewayResultCode.GW_RES_INVALID_PARAM

**Pull request merged**